### PR TITLE
move common code into server, update tests and makefile

### PR DESCRIPTION
**Pull request by MDunitz**
#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes

- modify, move common code into server, remove backend directory. Update tests and makefile

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -2,4 +2,3 @@ bin
 client
 dist
 docs
```
Comment (bkmartinjr) [inline .dockerignore:4]: does this need to have `server` added?  (I am not expert on our docker config)

File: Makefile [hunk 2]
```diff
@@ -33,22 +33,19 @@ build-client:
...
-	$(call copy_client_assets,client/build,backend/server)
+	$(call copy_client_assets,client/build,server)
 
 .PHONY: copy-client-assets
 copy-client-assets:
-	$(call copy_client_assets,client/build,backend/server)
+	$(call copy_client_assets,client/build,server)
 
 .PHONY: copy-client-assets-czi-hosted
 copy-client-assets-czi-hosted:
```
Comment (maniarathi) [inline Makefile:51]: Can be deleted?

File: test/locust/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Locust Load Test
 
-This directory contains scripts to load test cellxgene's backend. It
+This directory contains scripts to load test cellxgene's  It
```
Comment (bkmartinjr) [inline test/locust/README.md:3]: missing word...   I'm not sure you intended to delete this `backend` :-)

File: test/unit/converters/schema/test_validate.py [hunk 4]
```diff
@@ -408,27 +382,20 @@ def test_get_schema(self):
 
 
 class TestValidate(unittest.TestCase):
-
     def setUp(self):
-        self.source_h5ad_path = f"{PROJECT_ROOT}/backend/test/fixtures/pbmc3k-CSC-gz.h5ad"
+        self.source_h5ad_path = f"{PROJECT_ROOT}/test/fixtures/pbmc3k-CSC-gz.h5ad"
```
Comment (bkmartinjr) [inline test/unit/converters/schema/test_validate.py:386]: FIXTURES_ROOT?

**Review by bkmartinjr — CHANGES_REQUESTED** [review]
comments:

1. Do you also need to update the following files?
    * `.bumpversion.cfg`
    * `compatibility_tests.yml`
    * `push_tests.yml`

2. The following fails on my box (I think it used to work) - at top levek, run the following commands, and the last one (unit-test) fails:
    * make build-for-server-dev
    * pip install -e .
    * make unit-test 

     The unit test fails in the _client_ unit tests.  The actual cellxgene build works fine.

3. couple of other tiny nits inline

**Comment by MDunitz** [comment]
> comments:
> 
> 1. Do you also need to update the following files?
>    
>    * `.bumpversion.cfg`
>    * `compatibility_tests.yml`
>    * `push_tests.yml`

Yes
> 2. The following fails on my box (I think it used to work) - at top levek, run the following commands, and the last one (unit-test) fails:
>    
>    * make build-for-server-dev
>    * pip install -e .
>    * make unit-test
>    
>    The unit test fails in the _client_ unit tests.  The actual cellxgene build works fine.

Updated, they currently work locally for me but please let me know if they still dont work for you
> 3. couple of other tiny nits inline

File: test/locust/README.md (hunk 3 above)
Comment (bkmartinjr) [reply test/locust/README.md:3]: still not fixed

**Comment by bkmartinjr** [comment]
> > The unit test fails in the client unit tests. The actual cellxgene build works fine.
> 
> Updated, they currently work locally for me but please let me know if they still dont work for you

It still fails for me.  I did from top-level:
1. make clean
2. make build-for-server-dev
3. pip install -e .
4. make unit-test

It fails.  Is it related to #2428 ?


Errors:

```

Summary of all failing tests
 FAIL  __tests__/util/annoMatrix/annoMatrix.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: window is not defined

      1 | export const baseDataURL = "https://a.fake.url/api/v0.2";
      2 |
    > 3 | window.CELLXGENE = {
        | ^
      4 |   API: {
      5 |     prefix: baseDataURL,
      6 |     version: "v0.2/",

      at Object.<anonymous> (__tests__/util/annoMatrix/serverMocks/index.js:3:1)
      at Object.<anonymous> (__tests__/util/annoMatrix/annoMatrix.test.js:3:1)

 FAIL  __tests__/util/annoMatrix/crossfilter.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: window is not defined

      1 | export const baseDataURL = "https://a.fake.url/api/v0.2";
      2 |
    > 3 | window.CELLXGENE = {
        | ^
      4 |   API: {
      5 |     prefix: baseDataURL,
      6 |     version: "v0.2/",

      at Object.<anonymous> (__tests__/util/annoMatrix/serverMocks/index.js:3:1)
      at Object.<anonymous> (__tests__/util/annoMatrix/crossfilter.test.js:3:1)

 FAIL  __tests__/util/actionHelpers.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      3 | /** Singleton toaster instance. Create separate instances for different options. */
      4 |
    > 5 | const ToastTopCenter = Toaster.create({
        |                                ^
      6 |   className: "recipe-toaster",
      7 |   position: Position.TOP,
      8 |   maxToasts: 4,

      at Function.create (node_modules/@blueprintjs/core/src/components/toast/toaster.tsx:126:49)
      at Object.<anonymous> (src/components/framework/toasters.js:5:32)
      at Object.<anonymous> (src/util/actionHelpers.js:3:1)

 FAIL  __tests__/util/annoMatrix/whereCache.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      3 | /** Singleton toaster instance. Create separate instances for different options. */
      4 |
    > 5 | const ToastTopCenter = Toaster.create({
        |                                ^
      6 |   className: "recipe-toaster",
      7 |   position: Position.TOP,
      8 |   maxToasts: 4,

      at Function.create (node_modules/@blueprintjs/core/src/components/toast/toaster.tsx:126:49)
      at Object.<anonymous> (src/components/framework/toasters.js:5:32)
      at Object.<anonymous> (src/util/actionHelpers.js:3:1)

 FAIL  __tests__/util/centroid.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      3 | /** Singleton toaster instance. Create separate instances for different options. */
      4 |
    > 5 | const ToastTopCenter = Toaster.create({
        |                                ^
      6 |   className: "recipe-toaster",
      7 |   position: Position.TOP,
      8 |   maxToasts: 4,

      at Function.create (node_modules/@blueprintjs/core/src/components/toast/toaster.tsx:126:49)
      at Object.<anonymous> (src/components/framework/toasters.js:5:32)
      at Object.<anonymous> (src/util/actionHelpers.js:3:1)

 FAIL  __tests__/util/stateManager/colorHelpers.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      3 | /** Singleton toaster instance. Create separate instances for different options. */
      4 |
    > 5 | const ToastTopCenter = Toaster.create({
        |                                ^
      6 |   className: "recipe-toaster",
      7 |   position: Position.TOP,
      8 |   maxToasts: 4,

      at Function.create (node_modules/@blueprintjs/core/src/components/toast/toaster.tsx:126:49)
      at Object.<anonymous> (src/components/framework/toasters.js:5:32)
      at Object.<anonymous> (src/util/actionHelpers.js:3:1)

 FAIL  __tests__/reducers/genesets.test.js
  ● Test suite failed to run

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      3 | /** Singleton toaster instance. Create separate instances for different options. */
      4 |
    > 5 | const ToastTopCenter = Toaster.create({
        |                                ^
      6 |   className: "recipe-toaster",
      7 |   position: Position.TOP,
      8 |   maxToasts: 4,

      at Function.create (node_modules/@blueprintjs/core/src/components/toast/toaster.tsx:126:49)
      at Object.<anonymous> (src/components/framework/toasters.js:5:32)
      at Object.<anonymous> (src/util/actionHelpers.js:3:1)


Test Suites: 7 failed, 17 passed, 24 total
Tests:       213 passed, 213 total
Snapshots:   0 total
Time:        2.24 s, estimated 4 s
Ran all test suites.
make[1]: *** [Makefile:58: unit-test] Error 1
make[1]: Leaving directory '/home/bruce/projects/cellxgene/client'
make: *** [Makefile:62: unit-test-client] Error 2
```

**Review by bkmartinjr — APPROVED** [review]
LGTM

File: server/Makefile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+.PHONY: clean
+clean:
+	rm -f common/web/templates/index.html
+	rm -rf common/web/static
+	rm -f common/web/csp-hashes.json
+
+
+.PHONY: test-annotations-performance
+test-annotations-performance:
+	python ../test/performance/performance_test_annotations_py
```
Comment (atolopko-czi) [inline server/Makefile:12]: file extension: `.py`

File: server/__main__.py [hunk 6]
```diff
@@ -2,10 +2,13 @@
 if __package__ is None:
     import sys
     from pathlib import Path
+    import pdb
 
+    pdb.set_trace()
```
Comment (atolopko-czi) [inline server/__main__.py:7]: debug code to remove

File: server/Makefile (hunk 5 above)
Comment (MDunitz) [reply server/Makefile:12]:
```suggestion
	python ../test/performance/performance_test_annotations.py
```

File: server/__main__.py [hunk 7]
```diff
@@ -2,10 +2,13 @@
 if __package__ is None:
     import sys
     from pathlib import Path
+    import pdb
```
Comment (MDunitz) [inline server/__main__.py:5]:
```suggestion
```

File: server/__main__.py (hunk 6 above)
Comment (MDunitz) [reply server/__main__.py:7]:
```suggestion
```

**Review by atolopko-czi — APPROVED** [review]
LGTM. `cellxgene` install and runs, and unit tests pass for me (with the exception of `colorHelpers.test.js`, which I've never seen run to completion on my localhost!).

**Comment by atolopko-czi** [comment]
`Code scanning results / CodeQL` check fails, but it looks like it can be ignored since it's in test code?

**Pull request merged**